apps/emqx_dashboard/src/emqx_dashboard_monitor.erl

```diff
@@ -263,9 +263,9 @@ compact(Now, [{Time, Data} | Rest], Acc) ->
     NewAcc = merge_to_bucket(Bucket, Data, Acc),
     compact(Now, Rest, NewAcc).
 
-merge_to_bucket(Bucket, Data, [{Bucket, Data0} | Acc]) ->
-    NewData = merge_sampler_maps(Data, Data0),
-    [{Bucket, NewData} | Acc];
+%% Assumes input data is sorted chronologically.
```

Comment:
Table type is `set`, but it seems to be ordered when it reaches here...  Does it maintain insertion order?  Or is it just luck that `ets:select` returns thinks in the correct order?  :thinking:

Reply:
it's sorted by lists:keysort

Reply:
Not in all places: https://github.com/emqx/emqx/blob/e378169e5227f29b3b838cbceb0ef3eb131ee5da/apps/emqx_dashboard/src/emqx_dashboard_monitor.erl#L322-L325

I assume this place would also need this sort.